proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+
+        /**
+         * This Object type may change with the transport refactor
+         */
+        public Builder setTransportType(BaseTransportConfig transport){}
+
+        public Builder setContext(Context context){}
+
+        public SdlManager build() {}
+    }
+
+    // MANAGER GETTERS
+
+    public FileManager getFileManager() {}
+
+    public VideoStreamingManager getVideoStreamingManager() {}
+
+    public AudioStreamManager getAudioStreamManager() {}
+
+    public ScreenManager getScreenManager() {}
+
+    public LockscreenManager getLockscreenManager() {}
+
+    public PermissionManager getPermissionManager() {}
+
+    public SystemCapabilityManager getSystemCapabilityManager() {}
+
+    // SENDING REQUESTS
+
+    public void sendRPCRequest(RPCRequest request) throws SdlException {}
+
+    public void sendSequentialRequests(final List<? extends RPCRequest> rpcs, final OnMultipleRequestListener listener) throws SdlException {}
+
+    public void sendRequests(List<? extends RPCRequest> rpcs, final OnMultipleRequestListener listener) throws SdlException {}
+
+    // LIFECYCLE / OTHER
+
+    @Override
+    public void onProxyClosed(String info, Exception e, SdlDisconnectedReason reason){}
+
+    @Override
+    public void onServiceEnded(OnServiceEnded serviceEnded){}
+
+    @Override
+    public void onServiceNACKed(OnServiceNACKed serviceNACKed){}
+
+    @Override
+    public void onError(String info, Exception e){}
+}
+```
+
+#### BaseSubManager
+
+This is an abstract class that contains common lifecycle elements and methods between submanagers. This proposal only intends to include a small amount of code for this class, however, during implementation if logic is found to be common between different managers, it will be moved into this abstract class.
+
+```java
+public abstract class BaseSubManager {
+
+	// allows better handling of sub managers by maintaining their states
+	public enum ManagerState {
+		SETTING_UP,
+		READY,
+		SHUTDOWN,
+		ERROR,;
+	}
+
+	protected ISdl internalInterface;
+	protected ManagerState state;
+
+	public BaseSubManager(@NonNull ISdl internalInterface){
+		this.internalInterface = internalInterface;
+		this.state = ManagerState.SETTING_UP;
+	}
+
+	/**
+	 * <p>Called when manager is being torn down</p>
+	 */
+	public void dispose(){
+		this.state = ManagerState.SHUTDOWN;
+	}
+
+	protected void transitionToState(ManagerState state) {
+		this.state = state;
+	}
+
+	protected ManagerState getState() {
+		return state;
+	}
+}
+	
+```
+
+#### LockScreenManager (and default lock screen)
+
+The lockscreen manager maintains the base logic as the previous lockscreen manager, but also adds in key functionality that was missing previously. New items include a default lockscreen (layout and activity) and the ability to customize that default lockscreen layout. Apps are also able to pass in custom layouts or disable it completely and roll their own.
```

Comment:
Should the first sentence read "...maintains the same base logic as the previous lockscreen manager..." ?  Additionally, was there a previous Android manager, or should this clarify that it's referencing the iOS lockscreen manager?